cmake/switch/Switch.cmake

```diff
@@ -81,16 +81,22 @@ function(nx_generate_nacp outfile)
 
 	add_custom_command(
 		OUTPUT "${outfile}"
-		COMMAND "${NX_NACPTOOL_EXE}" --create "${NACP_NAME}" "${NACP_AUTHOR}" "${NACP_VERSION}" "${outfile}"
+		COMMAND "${NX_NACPTOOL_EXE}" --create "$<TARGET_FILE:${NACP_NAME}>" "${NACP_AUTHOR}" "${NACP_VERSION}" "${outfile}"
+		DEPENDS ${NACP_NAME}
 		VERBATIM
 	)
 endfunction()
 
-function(nx_create_nro prefix)
+function(nx_create_nro target)
 	cmake_parse_arguments(ELF2NRO "NOICON;NONACP" "ICON;NACP;ROMFS" "" ${ARGN})
 
-	set(ELF2NRO_ARGS ${prefix} "${prefix}.nro")
-	set(ELF2NRO_DEPS ${prefix})
+	get_target_property(TARGET_OUTPUT_NAME ${target} OUTPUT_NAME)
+	if(NOT TARGET_OUTPUT_NAME)
+		set(TARGET_OUTPUT_NAME "${target}")
+	endif()
```

Comment:
I'm wondering if it is possible to make this too into a generator expression. Preferably replacing the extension with .nro instead of outright appending when applicable.

Reply:
It's used as `OUTPUT` argument of `add_custom_command`.
When using
```cmake
		OUTPUT "$<TARGET_FILE_NAME:${target}>"
```
instead, the following error is emitted:
```
CMake Error at /conan/Switch.cmake:147 (add_custom_command):
  add_custom_command called with OUTPUT containing a "<".  This character is
  not allowed.
Call Stack (most recent call first):
  CMakeLists.txt:11 (nx_create_nro)


-- Configuring incomplete, errors occurred!

```

Reply:
Ah, so it just doesn't support generator expressions in this context. Kinda sucks, but understandable.